### PR TITLE
Closes #1680 enable renaming in by_vars

### LINKED ISSUE #1680: Feature Request:  Enable Renaming in `by_vars` Parameter
### Feature Idea

For all functions (where it makes sense) the `by_vars` argument should allow renaming like the dplyr `*_join()` functions and `derive_vars_joined()`.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

```
adtr <- derive_vars_merged(
  tr,
  dataset_add = tu,
  new_vars = vars(TULOC),
  by_vars = vars (STUDYID, USUBID, TRLNKID = TULINKID)
)
```

### PR DESCRIPTION
**Pull request by gg106046**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -43,9 +43,11 @@ that need carrying the last observation forward other than `analysis_var`
...
 observations or the first or last observation in a by group. (#1724)
 
-- `ADLB` metadata data set called `atoxgr_criteria_ctcv5` updated to remove unit check for
-`HYPERURICEMIA` as grade criteria based on `ANRHI` only.  This metadata holds criteria for lab grading
-based on [Common Terminology Criteria for Adverse Events (CTCAE) v5.0](https://ctep.cancer.gov/protocoldevelopment/electronic_applications/ctc.htm) (#1650)
+- The functions `derive_vars_merged`, `derive_var_merged_cat`, 
+`derive_var_merged_character`, `derive_var_merged_exist_flag`, 
+`derive_var_merged_summary`, and `derive_vars_merged_lookup` were updated to 
+allow renaming in the argument `by_vars` (#1680).
+
```
Comment (bms63) [inline NEWS.md:50]: Can you use () in all the functions.  This way the become links to the function documentation.

image

File: R/derive_merged.R [hunk 2]
```diff
@@ -61,7 +61,9 @@
 #'
 #'   The input dataset and the selected observations from the additional dataset
 #'   are merged by the specified by variables. The by variables must be a unique
-#'   key of the selected observations.
+#'   key of the selected observations. Variables from the additional dataset can
+#'   be renamed by naming the element, i.e., `by_vars =
+#'   exprs(<name in input dataset> = <name in additional dataset>)`.
```
Comment (bms63) [inline R/derive_merged.R:66]: Should we put something suggesting this is similar to the sql join or dplyr joins??  @bundfussr what do you think?

**Review by bms63 — COMMENTED** [review]
What do you all think about the messaging?  The first screenshot is from admiral and the second is from dplyr.  I think dplyr is more helpful - but we could implement this in a later release?  @zdz2101 @bundfussr 

image

image

File: renv.lock [hunk 3]
```diff
@@ -36,32 +36,6 @@
         "promises"
       ]
```
Comment (bms63) [inline renv.lock:37]: I'm worried about why this is updated??

File: tests/testthat/test-derive_merged.R [hunk 4]
```diff
@@ -455,3 +563,33 @@ test_that("derive_var_merged_summary Test 16: dataset != dataset_add, filter", {
     keys = c("USUBJID")
   )
 })
+
+## Test 21: by_vars with rename ----
+test_that("derive_var_merged_summary Test 21: by_vars with rename", {
```
Comment (bms63) [inline tests/testthat/test-derive_merged.R:568]: This is a really nice test!

**Review by bms63 — CHANGES_REQUESTED** [review]
* Could we get a test for `derive_var_merge_lookup` with the rename ability

* Could we have a few more tests that toggle on the `mode`, `filter_add`, `check_type` and `duplicate_msg` arguments?  Unless I missed them in the file.  I just want to make sure this update has no impact on the interactions of the arguments.

File: NEWS.md (hunk 1 above)
Comment (millerg23) [reply NEWS.md:50]: Was the section about ADLB grading update deliberately removed?

Comment (bms63) [reply NEWS.md:50]: I think it got moved down - something happened in a merge and grading update has been getting moved around in every PR :(

**Comment by bundfussr** [comment]
> What do you all think about the messaging for doing joins while renaming variables? The first screenshot is from admiral and the second is from dplyr. I think dplyr is more helpful - but we could implement this in a later release? @zdz2101 @bundfussr
> 
> image
> 
> image

@bms63 , I think both messages could be improved because they do not state in which dataset the variable is expected.

**Comment by bms63** [comment]
> > What do you all think about the messaging for doing joins while renaming variables? The first screenshot is from admiral and the second is from dplyr. I think dplyr is more helpful - but we could implement this in a later release? @zdz2101 @bundfussr
> > image
> > image
> 
> @bms63 , I think both messages could be improved because they do not state in which dataset the variable is expected.

Probably not something we should do right now.  Shall I make a new issue for a later update?

**Comment by bundfussr** [comment]
> > > What do you all think about the messaging for doing joins while renaming variables? The first screenshot is from admiral and the second is from dplyr. I think dplyr is more helpful - but we could implement this in a later release? @zdz2101 @bundfussr
> > > image
> > > image
> > 
> > 
> > @bms63 , I think both messages could be improved because they do not state in which dataset the variable is expected.
> 
> Probably not something we should do right now. Shall I make a new issue for a later update?

Yes, we would need to update `assert_data_frame()` and then update the calls in all functions with more than one input dataset.

File: R/derive_merged.R (hunk 2 above)
Comment (bundfussr) [reply R/derive_merged.R:66]: I think the description is pretty clear. But it would not harm to mention the similarity to dplyr.

Comment (zdz2101) [reply R/derive_merged.R:66]: Agreed, don't think there is harm in mentioning it, but I think it'll best be displayed/interpreted through an example.

File: NEWS.md (hunk 1 above)
Comment (gg106046) [reply NEWS.md]: Updated this. Not sure about the ADLB grading update. I'll look into it.

File: R/derive_merged.R (hunk 2 above)
Comment (gg106046) [reply R/derive_merged.R]: Updated the document to suggest that it's similar to dplyr joins. Do you want an example to be included as well that shows the renaming by_vars capability?

**Comment by gg106046** [comment]
> * Could we get a test for `derive_var_merge_lookup` with the rename ability
> * Could we have a few more tests that toggle on the `mode`, `filter_add`, `check_type` and `duplicate_msg` arguments?  Unless I missed them in the file.  I just want to make sure this update has no impact on the interactions of the arguments.

Added a test for `derive_var_merge_lookup` with reaming by_vars ability. There were unit tests present for `mode` already. Added a few more unit tests for `filter_add`, `check_type` and `duplicate_msg`.

**Review by bundfussr — APPROVED** [review]

**Review by bms63 — APPROVED** [review]

**Pull request merged**